### ISSUE TITLE
about mlflow.sklearn.***_model report an error

### ISSUE THREAD

**Issue by panxuexue** [mlflow/mlflow#183]
### Describe the problem
when I exec about mlflow.sklearn.log_model or mlflow.sklearn.save_model,will report an error

![2018-07-21 17-40-42](https://user-images.githubusercontent.com/41239150/43034363-b8068032-8d0d-11e8-989e-87bf994c6d16.png)

### Source code / logs
import numpy as np

from keras.models import Sequential
from keras.layers import Dense
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score
import mlflow.sklearn
import sys
reload(sys)
sys.setdefaultencoding('utf8')
np.random.seed(1337)  # for reproducibility

if __name__ == "__main__":

    X = np.linspace(-1, 1, 200)
    np.random.shuffle(X)  # randomize the data
    Y = 0.5 * X + 2 + np.random.normal(0, 0.05, (200,))
    # plot data
    # plt.scatter(X, Y)
    # plt.show()
    X_train, Y_train = X[:160], Y[:160]  # first 160 data points
    X_test, Y_test = X[160:], Y[160:]  # last 40 data points

    # build a neural network from the 1st layer to the last layer
    with mlflow.start_run():
        # build a neural network from the 1st layer to the last layer
        model = Sequential()
        model.add(Dense(output_dim=1, input_dim=1))

        # choose loss function and optimizing method
        loss = str(sys.argv[1]) if len(sys.argv) > 1 else 'mse'
        optimizer = str(sys.argv[2]) if len(sys.argv) > 2 else 'sgd'

        model.compile(loss=loss, optimizer=optimizer)

        # training
        print('Training -----------')
        for step in range(301):
            cost = model.train_on_batch(X_train, Y_train)
            if step % 100 == 0:
                print('train cost: ', cost)

        # test
        print('\nTesting ------------')
        cost = model.evaluate(X_test, Y_test, batch_size=40)
        print('test cost:', cost)
        W, b = model.layers[0].get_weights()
        print('Weights=', W, '\nbiases=', b)
        Y_pred = model.predict(X_test)

        rmse = np.sqrt(mean_squared_error(Y_test, Y_pred))

        mae = mean_absolute_error(Y_test, Y_pred)

        r2 = r2_score(Y_test, Y_pred)

        # plotting the prediction

        # plt.scatter(X_test, Y_test)
        #
        # plt.plot(X_test, Y_pred)

        # plt.show()

        print("Sequential model (loss=%s, optimizer=%s):" % (loss, optimizer))
        mlflow.log_param("loss", loss)
        mlflow.log_param("optimizer", optimizer)
        mlflow.log_metric("rmse", rmse)
        mlflow.log_metric("r2", r2)
        mlflow.log_metric("mae", mae)
        mlflow.log_metric("weight", W)
        mlflow.log_metric("biases", b)
        # mlflow.sklearn.log_model(model, "model")
        mlflow.sklearn.save_model(model, "/home/panxue/mm", conda_env=None, mlflow_model=None)

**Comment by juntai-zheng** [comment]
It seems like you're using the `sklearn` module from MLflow on a Keras Sequential model. The `sklearn` module is only compatible with `sklearn` models, so that's what's causing your error (more specifically, `sklearn.save_model()` and `sklearn_log_model()` attempt to pickle the input model, which is discouraged as seen [here](https://keras.io/getting-started/faq/#how-can-i-save-a-keras-model)). To use MLflow with your model, you can use MLflow's Tensorflow module for serving.

**Comment by smurching** [comment]
Thanks @juntai-zheng @panxuexue. Support for logging Keras models is being worked on in #280, once it's merged you should be able to log your model with MLflow :)
